code/notificationbuilder/src/main/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/BasicPushTemplate.kt

```diff
@@ -88,57 +87,47 @@ internal class BasicPushTemplate : AEPPushTemplate {
-        private set
+    internal val remindLaterTimestamp: Long?
 
     // Optional, If present, schedule this notification to be re-delivered after this provided time (in seconds).
-    internal var remindLaterDelaySeconds: Int?
-        private set
+    internal val remindLaterDuration: Int?
 
-    constructor(data: Map<String, String>) : super(data) {
-        actionButtonsString =
-            DataReader.optString(data, PushTemplateConstants.PushPayloadKeys.ACTION_BUTTONS, null)
+    /**
+     * Initializes the push template with the given NotificationData.
+     *
+     * @param data the data to initialize the push template with
+     * @param fromIntent flag to denote if the push template was built from an intent
+     */
+    init {
+        actionButtonsString = data.getString(PushPayloadKeys.ACTION_BUTTONS)
         actionButtonsList = getActionButtonsFromString(actionButtonsString)
-        remindLaterText = DataReader.optString(
-            data, PushTemplateConstants.PushPayloadKeys.REMIND_LATER_TEXT, null
-        )
-        val epochTimestampString = DataReader.optString(
-            data, PushTemplateConstants.PushPayloadKeys.REMIND_LATER_EPOCH_TIMESTAMP, null
-        )
-        remindLaterEpochTimestamp =
-            if (epochTimestampString.isNullOrEmpty()) null else epochTimestampString.toLong()
-
-        val delaySeconds = DataReader.optString(
-            data,
-            PushTemplateConstants.PushPayloadKeys.REMIND_LATER_DELAY_SECONDS, null
-        )
-        remindLaterDelaySeconds = if (delaySeconds.isNullOrEmpty()) null else delaySeconds.toInt()
+        remindLaterText = data.getString(PushPayloadKeys.REMIND_LATER_TEXT)
+        remindLaterTimestamp = data.getLong(PushPayloadKeys.REMIND_LATER_TIMESTAMP)
+        remindLaterDuration = data.getInteger(PushPayloadKeys.REMIND_LATER_DURATION)
     }
 
-    constructor(intent: Intent) : super(intent) {
-        val intentExtras =
-            intent.extras ?: throw IllegalArgumentException("Intent extras are null")
-        actionButtonsString =
-            intentExtras.getString(PushTemplateIntentConstants.IntentKeys.ACTION_BUTTONS_STRING)
-        actionButtonsList = getActionButtonsFromString(actionButtonsString)
-        remindLaterEpochTimestamp =
-            intentExtras.getLong(PushTemplateIntentConstants.IntentKeys.REMIND_EPOCH_TS)
-        remindLaterDelaySeconds =
-            intentExtras.getInt(PushTemplateIntentConstants.IntentKeys.REMIND_DELAY_SECONDS)
-        remindLaterText =
-            intentExtras.getString(PushTemplateIntentConstants.IntentKeys.REMIND_LABEL)
+    /**
+     * Creates an intent with the provided action and adds the remind later text, timestamp, duration and action buttons.
+     *
+     * @param action [String] containing the action to be added to the intent
+     * @return an [Intent] with the provided action and additional data
+     */
+    override fun createIntent(action: String): Intent {
```

Comment:
Same comment here and for subsequent template classes. I prefer keeping the intent creation in the builder rather than the template classes